src/core/binding.c

```diff
@@ -1422,23 +1423,25 @@ QuicBindingDeliverDatagrams(
     // packet, then the packet is dropped.
     //
 
-    QUIC_CONNECTION* Connection;
-    if (!Binding->ServerOwned || Packet->IsShortHeader) {
-        Connection =
-            QuicLookupFindConnectionByLocalCid(
-                &Binding->Lookup,
-                Packet->DestCid,
-                Packet->DestCidLen);
-    } else {
+    QUIC_CONNECTION* Connection = NULL;
+
+    if (Binding->ServerOwned) {
```

Comment:
You don't want to do this for short header packets. Only server long header packets.

Reply:
Also to better optimize the branches, I think following the old pattern, but adding another `ByLocalCid` in the case `ByRemoteHash` fails and there are client users.